lib/rules/max-lines.js

```diff
@@ -127,14 +138,28 @@ module.exports = {
                 if (skipComments) {
                     const comments = sourceCode.getAllComments();
 
-                    const commentLines = lodash.flatten(comments.map(comment => getLinesWithoutCode(comment)));
+                    const commentLines = lodash.flatten(
+                        comments.map(comment => getLinesWithoutCode(comment))
+                    );
 
-                    lines = lines.filter(l => !lodash.includes(commentLines, l.lineNumber));
+                    lines = lines.filter(
+                        l => !lodash.includes(commentLines, l.lineNumber)
+                    );
                 }
+                const loc = {
+                    start: {
+                        line: max + 1,
+                        column: 0
+                    },
+                    end: {
+                        line: lines.length,
+                        column: 0
+                    }
+                };
```

Comment:
If the settings are `{ "skipComments": true, "skipBlankLines": true }`, the line at index `max` doesn't have to be the first line above the maximum allowed lines.

Also, `lines` were already filtered at this point, so `lines.length` doesn't have to be the last line in the source code.

For example, the current iteration on an example with comments and empty lines would look like this :

![image](https://user-images.githubusercontent.com/44349756/82527163-0a34e780-9b36-11ea-8219-cfe087685e10.png)

Reply:
in latest commit, the following is the result 


![image](https://user-images.githubusercontent.com/26347874/82571586-caefb080-9ba0-11ea-929e-cf5d15b2a5a2.png)